File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        '''
        Input: temperatures = [73,74,75,71,69,72,76,73]
        Output: [1,1,4,2,1,1,0,0]
        
        1. init a stack and an ans list
        2. iterative the input list from 1 to end 
           
        
        '''
        
        res = [0] * len(temperatures)
        stack = []
        
        for i,t in enumerate(temperatures):
            while stack and t > stack[-1][0]:
                stackT, stackIdx = stack.pop()
                res[stackIdx] = i - stackIdx
            
            stack.append([t,i])
            
        return res

File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    /*
    [73,74,75,71,69,72,76,73]
           c   i
     
     if( arr[c] < arr[i]) ans.push(i-cur). cur++. break 
     if( arr[c] >= arr[i] )
    */
 let n = temperatures.length;
    let ans = new Array(n).fill(0);
    let stack = [];
    for(let i=0; i<n; i++){
        while(stack.length && temperatures[i] > temperatures[stack[stack.length-1]]){
            let idx = stack.pop();
            ans[idx] = i-idx
        }
        
        stack.push(i)
    }
    return ans
};